article/article.Rmd

```diff
@@ -0,0 +1,226 @@
+
+These functions are particularly useful for use with existing methods and models, especially for checking the robustness of results.
+
+```{r coerce, warning=FALSE, message=FALSE}
+tibble::tibble(min = as.Date(dates_expand, min),
+               max = as.Date(dates_expand, max),
+               median = as.Date(dates_expand, median),
+               mean = as.Date(dates_expand, mean),
+               modal = as.Date(dates_expand, modal),
+               random = as.Date(dates_expand, random))
+```
+
+## Additional functionality 
+
+Several other functions are also offered in the \pkg{messydates} package.
+
+For example, one can check various logical tests for messy date objects. `is_messydate()` tests whether the object inherits the `mdate` class. `is_intersecting()` tests whether there is any intersection between two messy dates. `is_element()` similarly tests whether a messy date can be found within a messy date range or set. `is_similar()` tests whether two dates contain similar components. `is_precise()` tests for whether date is precise.
+
+```{r logical, warning=FALSE}
+is_messydate(as_messydate("2001-01-01"))
+is_messydate(as.Date("2001-01-01"))
+is_intersecting(as_messydate("2001-01"), as_messydate("2001-01-01..2001-02-22"))
+is_intersecting(as_messydate("2001-01"), as_messydate("2001-02-01..2001-02-22"))
+is_element(as_messydate("2001-01-01"), as_messydate("2001-01"))
+is_element(as_messydate("2001-01-01"), as_messydate("2001-02"))
+is_similar(as_messydate("2001-06-02"), as_messydate("2001-02-06"))
+is_similar(as_messydate("2001-06-22"), as_messydate("2001-02-06"))
+is_precise(as_messydate("2001-06-02"))
+is_precise(as_messydate("2001-02"))
+```
+
+Additionally, one can perform intersection (`md_intersect()`) and union (`md_union()`) on, inter alia, messy date class objects. Or 'join' that retains all elements, even if duplicated, with `md_multiset`.
+
+```{r set, warning=FALSE}
+md_intersect(as_messydate("2001-01-01..2001-01-20"),as_messydate("2001-01"))
+md_union(as_messydate("2001-01-01..2001-01-20"),as_messydate("2001-01"))
+md_multiset(as_messydate("2001-01-01..2001-01-20"),as_messydate("2001-01"))
+```
+
+As well, some arithmetic operations are available for messydates. For instance, one can add or subtract one year to all messy dates in a vector.
+
+```{r operate, warning=FALSE, message=FALSE}
+tibble::tibble(date = dates_expand, add = dates_expand + "1 day", subtract = dates_expand - "1 year")
+```
+
+## Case Study - 2001 Battles
+
+Dates, even for some recent events, can be messy. Take, for example, the dates of battles in 2001 according to [Wikipedia](https://en.wikipedia.org/wiki/List_of_battles_in_the_21st_century).
+The dates of these battles are often uncertain, with different levels of date precision being reported. The dataset is included in \pkg{messydates}.
```

Comment:
I got the impression that 'uncertain' here is associated with 'different levels of date precision', but this wasn't the case when the annotations were explained above (uncertain dates can be precise but are uncertain because they are derived from unreliable sources). Maybe it would be clearer to say dates are 'messy' here and then elaborate on why (eg. imprecise, unspecified components, uncertain etc)?